crates/environ/src/component/translate.rs

```diff
@@ -375,14 +375,31 @@ enum ClosedOverModule {
     Upvar(ModuleUpvarIndex),
 }
 
+/// The data model for objects that are not unboxed in locals.
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub enum LocalDataModel {
+    /// Data is stored in GC objects.
+    Gc {
+        /// The type index of the core GC types signature.
+        core_type: Option<ModuleInternedTypeIndex>,
```

Comment:
Possible alternative to this, and a number of other locations, could `Gc` be empty? The type of a function is something that's always known regardless of whether it's gc or linear memory, so could the type be adjacent to the "function item" and `Gc` here be empty? That way in the gc case there's still always a type available but it would also be available in the linear memory case too in case it's needed. (e.g. elsewhere when calculating the lowered signature we could avoid reimplementing algorithms in wasmtime I think?)